src/Features/CSharp/Portable/InlineHints/CSharpInlineTypeHintsService.cs

```diff
@@ -75,11 +77,19 @@ public CSharpInlineTypeHintsService(IGlobalOptionService globalOptions)
             {
                 if (node is ParameterSyntax { Type: null } parameterNode)
                 {
+                    var span = new TextSpan(parameterNode.Identifier.SpanStart, 0);
                     var parameter = semanticModel.GetDeclaredSymbol(parameterNode, cancellationToken);
                     if (parameter?.ContainingSymbol is IMethodSymbol { MethodKind: MethodKind.AnonymousFunction } &&
                         IsValidType(parameter?.Type))
                     {
-                        return new(parameter.Type, new TextSpan(parameterNode.Identifier.SpanStart, 0), trailingSpace: true);
+                        if (parameterNode.Parent!.Parent.IsKind(SyntaxKind.ParenthesizedLambdaExpression))
```

Comment:
Would we everr have a null Parent in this case?

Reply:
i would jsut write this as:

```c#
var textChange = parameterNode.Parent?.Parent?.Kind is SyntaxKind.ParenthesizedLambdaExpression
    ? new TextChange(...)
    : null;
```

then just return once.